src/Microsoft.Health.Fhir.Liquid.Converter/Filters/StringFilters.cs

```diff
@@ -12,39 +21,88 @@ namespace Microsoft.Health.Fhir.Liquid.Converter
     /// </summary>
     public partial class Filters
     {
-        public static char CharAt(string dataString, int index)
+        public static char CharAt(string data, int index)
         {
-            return dataString[index];
+            return data[index];
         }
 
         public static bool Contains(string parentString, string childString)
         {
-            if (parentString == null)
+            return parentString != null && parentString.Contains(childString);
+        }
+
+        public static string EscapeSpecialChars(string data)
+        {
+            return string.IsNullOrEmpty(data) ? data : SpecialCharProcessor.Escape(data);
+        }
+
+        public static string UnescapeSpecialChars(string data)
+        {
+            return string.IsNullOrEmpty(data) ? data : SpecialCharProcessor.Unescape(data);
+        }
+
+        public static List<string> Match(string data, string regexString)
+        {
+            var result = new List<string>();
+            if (string.IsNullOrEmpty(data))
             {
-                return false;
+                return result;
             }
 
-            return parentString.Contains(childString);
+            var regex = new Regex(regexString);
+            var matches = regex.Match(data);
+            foreach (Match match in matches.Captures)
+            {
+                result.Add(match.Value);
+            }
+
+            return result;
         }
 
-        public static string EscapeSpecialChars(string data)
+        public static string ToJsonString(object data)
         {
-            if (string.IsNullOrEmpty(data))
+            return data == null ? null : JsonConvert.SerializeObject(data, Formatting.None);
+        }
+
+        public static string Gzip(string data)
+        {
+            using var inputStream = new MemoryStream(Encoding.UTF8.GetBytes(data));
+            using var outputStream = new MemoryStream();
+            using (var gzipStream = new GZipStream(outputStream, CompressionMode.Compress))
             {
-                return data;
+                inputStream.CopyTo(gzipStream);
             }
 
-            return SpecialCharProcessor.Escape(data);
+            return Convert.ToBase64String(outputStream.ToArray());
         }
 
-        public static string UnescapeSpecialChars(string data)
+        public static string Gunzip(string data)
```

Comment:
function name is wired not all strings can be unzipped,  change to GunzipBase64String?